File: SysAudit-VB/Classes/WMI/Win32_NetworkAdapterConfiguration.vb
Public Class Win32_NetworkAdapterConfiguration
    Public Property Caption As String
    Public Property DefaultIpGateway As String()
    Public Property Description As String
    Public Property DHCPEnabled As Boolean
    Public Property DHCPServer As String
    Public Property DNSDomain As String
    Public Property DNSDomainSuffixSearchOrder As String()
    Public Property DNSEnabledForWINSResolution As Boolean
    Public Property DNSHostName As String
    Public Property DNSServerSearchOrder As String()
    Public Property DomainDNSRegistrationEnabled As Boolean
    Public Property FullDNSRegistrationEnabled As Boolean
    Public Property GatewayCostMetric As UShort()
    Public Property Index As UInteger
    Public Property InterfaceIndex As UInteger
    Public Property IPAddress As String()
    Public Property IPConnectionMetric As UInteger
    Public Property IPEnabled As Boolean
    Public Property IPFilterSecurityEnabled As Boolean
    Public Property IPSubnet As String()
    Public Property MACAddress As String
    Public Property ServiceName As String
    Public Property SettingID As String
    Public Property WINSEnableLMHostsLookup As Boolean
End Class

File: SysAudit/SysAudit/Classes/WMI/Win32_NetworkAdapterConfiguration.cs
namespace SysAudit.Classes.WMI
{
    public sealed class Win32_NetworkAdapterConfiguration
    {
        public string Caption { get; set; }
        public string[] DefaultIpGateway { get; set; }
        public string Description { get; set; }
        public bool DHCPEnabled { get; set; }
        public string DHCPServer { get; set; }
        public string DNSDomain { get; set; }
        public string[] DNSDomainSuffixSearchOrder { get; set; }
        public bool DNSEnabledForWINSResolution { get; set; }
        public string DNSHostName { get; set; }
        public string[] DNSServerSearchOrder { get; set; }
        public bool DomainDNSRegistrationEnabled { get; set; }
        public bool FullDNSRegistrationEnabled { get; set; }
        public ushort[] GatewayCostMetric { get; set; }
        public uint Index { get; set; }
        public uint InterfaceIndex { get; set; }
        public string[] IPAddress { get; set; }
        public uint IPConnectionMetric { get; set; }
        public bool IPEnabled { get; set; }
        public bool IPFilterSecurityEnabled { get; set; }
        public string[] IPSubnet { get; set; }
        public string MACAddress { get; set; }
        public string ServiceName { get; set; }
        public string SettingID { get; set; }
        public bool WINSEnableLMHostsLookup { get; set; }
    }
}